### PR TITLE
[BUGFIX]: TCA Exception for content elements in TYPO3 8.5.x. 

### PR DESCRIPTION
**Pull request by codekiste**
Moved dynamic created TCA from ext_tables.php to TCA/Overrides tt_content and pages

In relation to: https://forge.typo3.org/issues/78957

### CODE REVIEW

File: Configuration/TCA/Overrides/pages.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+$storageRepository = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('MASK\\Mask\\Domain\\Repository\\StorageRepository');
+$configuration = $storageRepository->load();
+
+if (!empty($configuration)) {
```
Comment (IchHabRecht) [inline Configuration/TCA/Overrides/pages.php]: should imho test at least for $configuration['pages']

File: Configuration/TCA/Overrides/tt_content.php [hunk 2]
```diff
@@ -9,4 +9,20 @@
...
 // and set mask itemsProcFuncs
 $GLOBALS['TCA']['tt_content']['columns']['colPos']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\ColPosList->itemsProcFunc';
-$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\CTypeList->itemsProcFunc';
\ No newline at end of file
+$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\CTypeList->itemsProcFunc';
+
+$storageRepository = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('MASK\\Mask\\Domain\\Repository\\StorageRepository');
+$configuration = $storageRepository->load();
+
+if (!empty($configuration)) {
```
Comment (IchHabRecht) [inline Configuration/TCA/Overrides/tt_content.php]: should IMHO test for $configuration['tt_content']

File: Configuration/TCA/Overrides/pages.php (hunk 1 above)
Comment (codekiste) [inline Configuration/TCA/Overrides/pages.php]: Thx, change commited

File: Configuration/TCA/Overrides/tt_content.php (hunk 2 above)
Comment (codekiste) [inline Configuration/TCA/Overrides/tt_content.php]: Thx, change commited

File: Configuration/TCA/Overrides/pages.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+$storageRepository = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('MASK\\Mask\\Domain\\Repository\\StorageRepository');
+$configuration = $storageRepository->load();
+
+if (!empty($configuration) && array_key_exists('pages', $configuration)) {
```
Comment (IchHabRecht) [inline Configuration/TCA/Overrides/pages.php]: sorry, would you mind to test with !empty($configuration['pages'])? array_key_exists() returns also true for $configuration['pages'] = null

File: Configuration/TCA/Overrides/tt_content.php [hunk 4]
```diff
@@ -9,4 +9,21 @@
...
 // and set mask itemsProcFuncs
 $GLOBALS['TCA']['tt_content']['columns']['colPos']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\ColPosList->itemsProcFunc';
-$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\CTypeList->itemsProcFunc';
\ No newline at end of file
+$GLOBALS['TCA']['tt_content']['columns']['CType']['config']['itemsProcFunc'] = 'MASK\Mask\ItemsProcFuncs\CTypeList->itemsProcFunc';
+
+$storageRepository = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance('MASK\\Mask\\Domain\\Repository\\StorageRepository');
+$configuration = $storageRepository->load();
+
+if (!empty($configuration) && array_key_exists('tt_content', $configuration)) {
```
Comment (IchHabRecht) [inline Configuration/TCA/Overrides/tt_content.php]: see above

**Comment by matin73** [comment]
hello,
there is still tca error on section (wiederholung) use:
There is no entry in the $TCA array for the table "tx_mask_loop". This means that the function enableFields() is called with an invalid table name as argument.

**Comment by IchHabRecht** [comment]
Unfortunately this was reintroduced with commit https://github.com/Gernott/mask/commit/43861ecd5828c58ef22b7d92fe65722dfe3a882f

**Comment by matin73** [comment]
but not solved yet, right?

**Comment by matin73** [comment]
strange is, that in firefox sometimes it works and there is no error ... if i open same page in ie or chrome the error is shown ...!?

**Comment by butu** [comment]
There was a reason why I moved it to ext_tables.php. It fixed some behaviour / bug when i did it. But apparently it was not the right decision, and it reintroduced this bug. I'm aware of it and will undo / fix this in the following days.

**Comment by butu** [comment]
Now it should be working. The actual TCA is now back in TCA/Overrides/tt_content.php
But the method call \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::allowTableOnStandardPages('tx_mask_inlinetable') still needs to be done in ext_tables.php. This is according to this documentation https://docs.typo3.org/typo3cms/TCAReference/Ctrl/Index.html and the error message I got everytime I tried to save an inline element, when everything was moved into the Overrides folder.

And I don't know why, but I had only 1 project where I could reproduce this exception in the frontend.  All the other (8.7) installations worked even without this fix.

**Comment by christian-forgacs** [comment]
+1 when came the merge and release in TER?

We have the same issue with the TCA generation...

**Comment by Gernott** [comment]
Implemented directly.

**Pull request closed without merging**